app/helpers/sessions_helper.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+module SessionsHelper
+  def sign_in(user)
+    remember_token = User.new_remember_token
+    cookies.permanent[:remember_token] = remember_token
+    user.update_attribute(:remember_token, User.digest(remember_token))
+    self.current_user = user
+  end
+
+  def current_user=(user)
+    @current_user = user
+  end
+
+  def current_user
+    remember_token = User.digest(cookies[:remember_token])
```

Comment:
What if you just use 
```ruby
cookies.permanent[:remember_token]
```

Reply:
I think we did that. In line 5 we create the cookie when the user signs in, and in line 6 using the cookies.permanent[:remember_token], we assign that very cookie we created. If there is a better or another way of doing it, please kindly expound more on that.

Reply:
I think the `permanent` method is needed just when you are going to assign a value on it, but if you just need to retrieve the value you can use `cookies[]` as shown here: https://www.learnenough.com/ruby-on-rails-4th-edition-tutorial/advanced_login#code-persistent_current_user.